rclpy/rclpy/validate_parameter_name.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from rclpy.exceptions import InvalidParameterException
+from rclpy.impl.implementation_singleton import rclpy_implementation as _rclpy
+
+
+def validate_parameter_name(name):
+    """
+    Validate a given parameter name, and raise an exception if invalid.
+
+    The name does not have to be fully-qualified and is not expanded.
+
+    If the name is invalid then rclpy.exceptions.InvalidParameterNameException
+    will be raised.
+
+    :param name str: topic or service name to be validated
+    :param is_service bool: if true, InvalidServiceNameException is raised
+    :returns: True when it is valid
+    :raises: InvalidParameterNameException: when the name is invalid
+    """
+    # TODO(jubeira): define whether this is an appropriate method to validate
+    # parameter names.
+    result = _rclpy.rclpy_get_validation_error_for_topic_name(name)
+    if result is None:
+        return True
+    error_msg, invalid_index = result
+    raise InvalidParameterException(name, error_msg, invalid_index)
```

Comment:
I do agree that `InvalidParameterName` exception is a better name, but in `rclcpp` the exception for invalid name is `InvalidParameterException`, and that's why I'm using that name here as well.

For now I changed the documentation to match this exception here https://github.com/ros2/rclpy/pull/325/commits/b56a25931f252123348348a71be3e1d0233ff396. I'm slightly inclined to using the same name as in `rclcpp` in this case, but I wouldn't mind refactoring it to `InvalidParameterNameException` if you think it's a better option.

Reply:
That's okay with me, but in `rclcpp` it's plural: `InvalidParametersExeception` :wink:

Reply:
Yes, indeed; I did notice that. I used singular because:
- All the remaining exceptions but this one are singular.
- The exception is only raised when one parameter name in a given list is incorrect when the name is verified; the verification isn't applied to all the parameters at once. This seems to match the [documentation](https://github.com/ros2/rclcpp/blob/a8a0788f81ddd70210df0c38b4434901d3a9ca44/rclcpp/include/rclcpp/node.hpp#L364-L365) for `rclcpp`.

That being said, perhaps we should just stick to the plural right?

Reply:
We could change the one in rclcpp if singular makes sense. I don't remember having a strong opinion about that. I probably used singular because at the time it was always thrown from `set_parameters` or `set_parameters_atomically` which were plural as well.

Reply:
Singular makes sense to me. We could update the rclcpp name for consistency.

Reply:
Alright, then let's keep it `InvalidParameterException` here :+1: